src/testing/AddTestToIdentifierPage.test.tsx

```diff
@@ -14,77 +15,140 @@ jest.mock('../common/useConfig');
   beforeEach(() => {
     mockConfigForEstonianIdMethodAndEnglish();
-    mockToken('some-token');
+    mockAuthentication();
   });
 
   it('focuses identifier input', async () => {
-    const { getByLabelText } = renderWrapped(<AddTestToIdentifierPage />);
+    mockHttp().get('/api/v1/test-types').reply(200, [aTestType()]);
 
-    const identifierInput = getByLabelText(/identification code/i);
+    renderWrapped(<AddTestToIdentifierPage />);
 
+    const identifierInput = screen.getByLabelText(/identification code/i);
     expect(identifierInput).toHaveFocus();
   });
 
-  it('creates a user for the authentication method from config and the filled identifier', async () => {
-    const authenticationDetails = { method: 'ESTONIAN_ID', identifier: '39210030814' };
-    const scope = nock(/./)
-      .post('/api/v1/users', { authenticationDetails })
-      .matchHeader('Authorization', 'Bearer some-token')
-      .reply(201, { id: 'some-id', authenticationDetails });
+  it('creates a user for the authentication method from config and the filled identifier and adds a test for that user', async () => {
+    mockHttp().get('/api/v1/test-types').reply(200, [aTestType()]);
 
-    const { getByText, getByLabelText, queryByText, findByText } = renderWrapped(
-      <AddTestToIdentifierPage />
-    );
+    renderWrapped(<AddTestToIdentifierPage />);
 
-    const identifierInput = getByLabelText(/identification code/i) as HTMLInputElement;
-    const submit = () => fireEvent.click(getByText(/add test result/i));
+    const identifierInput = screen.getByLabelText(/identification code/i);
+    const button = screen.getByRole('button');
+    const submit = () => userEvent.click(button);
 
-    fillInput(identifierInput, '79210030814'); // invalid id code
-    submit();
-    await findByText(/check the identification code/i); // validation error
-
-    fillInput(identifierInput, '39210030814'); // valid id code
-    await waitForElementToBeRemoved(queryByText(/check the identification code/i)); // no validation error
+    const negativeOption = await screen.findByRole('radio', { name: 'Negative' });
+    const positiveOption = screen.getByRole('radio', { name: 'Positive' });
+    const notesInput = screen.getByRole('textbox', { name: /notes/ });
 
+    userEvent.type(identifierInput, '79210030814'); // invalid id code
     submit();
-    await findByText(/39210030814 added/i);
-    expect(identifierInput.value).toBe('');
+    await screen.findByText(/check the identification code/i); // validation error
+
+    userEvent.type(identifierInput, '39210030814'); // valid id code
+    await waitForElementToBeRemoved(screen.queryByText(/check the identification code/i)); // no validation error
+
+    await userEvent.type(notesInput, 'Some notes');
+    expect(button).toBeDisabled(); // as a radio option has not been selected
+
+    userEvent.click(negativeOption);
+    userEvent.click(positiveOption);
+    await waitFor(() => expect(button).not.toBeDisabled());
```

Comment:
I was unclear here.  Firstly, why do you need to click both options?  Then I wasn't sure if button was a radio button without going up to check.

Reply:
This is unclear indeed — I had a bug where none of the changes after the first had any effect, so this part of the test ensures it won't happen again.

I'll rename `button` to `submitButton`, should I also rename `*Option`s to `*Radio` or `*RadioOption`?

Reply:
Added a comment and it's in a shorter test now, so hopefully clear.